File: Day29/Solution.cpp
int main(){
    int t;
    cin >> t;
    for(int a0 = 0; a0 < t; a0++){
        int n;
        int k;
        cin >> n >> k;
        int max = 0;
            for(int i=1;i<n;i++){
                for(int j=i+1;j<=n;j++){
                    if ((i & j) < k && (i & j) > max){
                        max = i&j;
                    }
                }
            }
        cout<<max<<endl;
    }
    return 0;
}

File: Day29/Solution.swift
import Foundation

let testCases = Int(readLine() ?? "") ?? 0
var n = 0
var k = 0

for _ in 0..<testCases {
    let input = readLine()?.split(separator: " ").map {Int($0)}
    n = input?[0] ?? 0
    k = input?[1] ?? 0
    var max = 0
    for x in 1..<n {
        for y in x+1...n {
            if ( x & y) < k && ( x & y) > max {
                max = x & y
            }
        }
    }
    print(max)
}
